src/lexer.coffee

```diff
@@ -246,7 +246,7 @@ exports.Lexer = class Lexer
         [poppedToken[2].first_line, poppedToken[2].first_column, poppedToken[2].range[0]]
     if colon
       colonOffset = input.lastIndexOf if inJSXTag then '=' else ':'
-      colonToken = @token ':', ':', offset: colonOffset, length: colon.length
+      colonToken = @token ':', ':', offset: colonOffset
```

Comment:
This was incorrectly assigning location data of eg length 2 to object colon tokens that were preceded by spaces (eg `a : b`)